appinventor/appengine/src/com/google/appinventor/server/FileImporterImpl.java

```diff
@@ -175,14 +175,17 @@ public long importFile(String userId, long projectId, String fileName,
     while ((bytes = bis.read(buffer, 0, buffer.length)) != -1) {
       bos.write(buffer, 0, bytes);
       fileLength += bytes;
+      if (fileLength > maxSizeBytes) {
+        // Read the rest of the stream, but throw it away
+        // and throw an error, so we do not consume memory storing
+        // a large object
+        while((bytes = bis.read(buffer, 0, buffer.length)) != -1) {
+        }
+        throw new FileImporterException(UploadResponse.Status.FILE_TOO_LARGE);
```

Comment:
Is it really necessary to continue reading the stream at this point? It seems like a waste of time, especially if the client is on a slower, possibly metered, connection.

Reply:
No. That's why this is a draft. We should do a better job in a real change. This change was intentionally minimalistic because we are going from development to production deployment in a matter of hours. The less we change, the better the chances we do not break things by finding new corner cases, either in our code or in the code we call.

Reply:
Understood.